app/main/routes.py

```diff
@@ -708,3 +711,179 @@ def terms_of_use():
+
+    if file is None:
+        abort(404)
+
+    filename = file.FileName
+
+    image = Image.open(io.BytesIO(s3_file_object["Body"].read()))
+    width, height = image.size
+
+    # Get the file from S3 to read dimensions
+    s3 = boto3.client("s3")
+    bucket = current_app.config["RECORD_BUCKET_NAME"]
+    key = f"{file.consignment.ConsignmentReference}/{file.FileId}"
+
+    s3_response_object = s3.get_object(Bucket=bucket, Key=key)
+    file_content = s3_response_object["Body"].read()
+    image = Image.open(io.BytesIO(file_content))
+    width, height = image.size
+
+    file_url = url_for("main.get_file", record_id=record_id, _external=True)
+
+    manifest = {
+        "@context": "http://iiif.io/api/presentation/2/context.json",
+        "@id": f"{url_for('main.get_file', record_id=record_id, _external=True)}",
+        "@type": "sc:Manifest",
+        "label": filename,
+        "description": f"Manifest for {filename}",
+        "sequences": [
+            {
+                "@id": f"{url_for('main.get_file', record_id=record_id, _external=True)}",
+                "@type": "sc:Sequence",
+                "canvases": [
+                    {
+                        "@id": f"{url_for('main.get_file', record_id=record_id, _external=True)}",
+                        "@type": "sc:Canvas",
+                        "label": "Image 1",
+                        "width": width,
+                        "height": height,
+                        "images": [
+                            {
+                                "@id": f"{url_for('main.get_file', record_id=record_id, _external=True)}",
+                                "@type": "oa:Annotation",
+                                "motivation": "sc:painting",
+                                "resource": {
+                                    "@id": file_url,
+                                    "type": "dctypes:Image",
+                                    "format": "image/png",
+                                    "width": width,
+                                    "height": height,
+                                },
+                                "on": f"{url_for('main.get_file', record_id=record_id, _external=True)}",
+                            }
+                        ],
+                    }
+                ],
+            }
+        ],
+    }
+
+    return jsonify(manifest)
+
+
+@bp.route("/files/<record_id>")
+def get_file(record_id=None):
+    file = db.session.get(File, record_id)
+    if file is None:
+        abort(404)
+
+    filename = file.FileName
+    s3 = boto3.client("s3")
+    bucket = current_app.config["RECORD_BUCKET_NAME"]
+    key = f"{file.consignment.ConsignmentReference}/{file.FileId}"
+
+    file_type = filename.split(".")[-1].lower()
+
+    try:
+        s3_response_object = s3.get_object(Bucket=bucket, Key=key)
+        file_content = s3_response_object["Body"].read()
+        file_type = filename.split(".")[-1].lower()
+
+        return send_file(
```

Comment:
is this different from the download route?

Reply:
Changed the download route to include the changes for render. An request arg is used to differentiate between needing to download the file as an attachment or just return the file for render